build_runner/lib/src/asset_graph/graph.dart

```diff
@@ -350,18 +355,23 @@ class AssetGraph {
   /// removed from the graph.
   Set<AssetId> _addGeneratedOutputs(Iterable<AssetId> outputs, int phaseNumber,
       BuilderOptionsAssetNode builderOptionsNode,
-      {AssetId primaryInput, @required bool isHidden}) {
+      {AssetId primaryInput,
+      @required bool isHidden,
+      @required bool allowDeclaredOutputConflicts}) {
     var removed = new Set<AssetId>();
     for (var output in outputs) {
       // When any outputs aren't hidden we can pick up old generated outputs as
       // regular `AssetNode`s, we need to delete them and all their primary
       // outputs, and replace them with a `GeneratedAssetNode`.
       if (contains(output)) {
         var node = get(output);
-        if (node is GeneratedAssetNode) {
+        if (allowDeclaredOutputConflicts) {
```

Comment:
should this also check whether it's a source node?

Reply:
This is actually an old check that I am removing in my next commit to fix a few tests, but good catch!